src/base.jl

```diff
@@ -661,6 +661,17 @@ function remove_component!(
     handle_component_removal!(sys, component)
 end
 
+"""
+Check to see if the component of type T with name exists. If T is an abstract
+type then the names of components across all subtypes of T must be unique.
+
+Throws ArgumentError if T is not a concrete type and there is more than one component with
+    requested name
```

Comment:
This is not really an error, so I think it would be better to return true in this case.